### PR TITLE
workaround for gledopto lights to recall scenes with ct

### PR DESCRIPTION
**Pull request by pSpitzner**
This is a naiv workaround for gledopto extended color lights, as described in #2364 

Gledopto extended color lights support setting color either as temperature `ct` or as `xy`.
Only if colors are set via `ct` will white leds be used in devices that support them. Otherwise, RGB leds are used.
This is problematic with scenes that (are supposed to) recall colors as temperatures.

Tested with a gledopto lightstrip, `GL-MC-001`, `Extended color light`

### CODE REVIEW

**Comment by domoritz** [comment]
That's awesome. Thank you for working on a fix.

**Comment by Schorsch980** [comment]
Great news! Can't wait to see this merged! I was close to revert to the Hue Bridge... not anymore :)

File: rest_groups.cpp [hunk 1]
```diff
@@ -2953,7 +2953,11 @@ int DeRestPluginPrivate::recallScene(const ApiRequest &req, ApiResponse &rsp)
                 else if(ls->colorMode() == QLatin1String("ct"))
                 {
                     item = lightNode->item(RStateCt);
-                    if (item && ls->colorTemperature() != item->toNumber())
+                    if (lightNode->manufacturer() == QLatin1String("GLEDOPTO") &&
+                        lightNode->type() == QLatin1String("Extended color light") ) {
+                        // pass, we update manually below
+                    }
```
Comment (manup) [inline rest_groups.cpp:2959]: For consistency please stick to to the remaining code style

```
if (lightNode->manufacturer() == QLatin1String("GLEDOPTO") &&
    lightNode->type() == QLatin1String("Extended color light"))
{
    // pass, we update manually below
}
```

File: rest_groups.cpp [hunk 2]
```diff
@@ -2986,6 +2990,25 @@ int DeRestPluginPrivate::recallScene(const ApiRequest &req, ApiResponse &rsp)
                 }
             }
 // #endif
+
+            if (lightNode->manufacturer() == QLatin1String("GLEDOPTO") &&
+                lightNode->type() == QLatin1String("Extended color light") &&
+                ls->colorMode() == QLatin1String("ct") ) {
+
```
Comment (manup) [inline rest_groups.cpp:2997]: Same here

```
if (lightNode->manufacturer() == QLatin1String("GLEDOPTO") &&
    lightNode->type() == QLatin1String("Extended color light") &&
    ls->colorMode() == QLatin1String("ct"))
{
}
```

**Review by manup — CHANGES_REQUESTED** [review]
Sorry took a while for reviewing the PR.

To be frank I'm not too thrilled by converting group casts into a series of unicasts since this can fill the APS queue surprisingly quickly. Later on a better scene controller will introduced to hopefully fix such cases as GLEDOPTO and IKEA in a more solid way https://github.com/dresden-elektronik/deconz-rest-plugin-v2/issues/5.

But I think this PR can be merged as workaround until the new scene controller arrives. I assume that the impact is not too bad since likely networks don't contain large numbers GLEDOPTO lights.

**Comment by torbenmb** [comment]
Is there any plan to merge this workaround in the release of deconz?
I would highly appreciate this as it might solve all issues I currently have with GLEDOPTO lights.

**Comment by arneeichhorn** [comment]
Are there any news on this?

BTW: GL-S-007P and GL-G-001P seem to support Zigbee Scenes in my setup.
GL-C-008 doesn't.
Maybe it would be possible to set a list of model numbers on a settings page or something for which devices to use unicast calls?
So that devices like the GL-S-007P GU10 and GL-G-001P garden spots still can use the more performant Zigbee scenes?

**Comment by realwax** [comment]
Hi,
I add myself to the request list of this fix as I intend to use the 2id glc008, too. I haven't found any other 5 channel rgb with cw and ww controller yet, that seems to be a better solution. Gledopto seems to be the only one vastly available.

**Comment by NathanSweet** [comment]
I'm stuck with a Gledopto bulb that I can't use with DeCONZ.

**Comment by pSpitzner** [comment]
Sorry for the long delay, I missed the github notifications.

I have fetched the upstream and followed the style of the ikea workaround [#4846](https://github.com/dresden-elektronik/deconz-rest-plugin/pull/4846). This seems much cleaner and easier to remove once a better fix is available.

**Comment by poelzi** [comment]
I was wondering why the gledopto light in general don't use the white leds for colors that are not fully saturated. The full part of the color should be substracted from the rgb part and mixed with both white leds. This makes much, much nicer colors. I wonder if this is a general problem with gledopto lights currently.

**Comment by MaxGramser** [comment]
@Mimiix is there an update about this?

**Comment by Mimiix** [comment]
@MaxGramser I don't do anything with pull requests. All I can do is forward it as @manup handles those

**Pull request merged**